src/libsyntax_ext/deriving/cmp/eq.rs

```diff
@@ -54,7 +54,7 @@ fn cs_total_eq_assert(cx: &mut ExtCtxt<'_>,
         // set the expn ID so we can use the unstable struct.
         let span = span.with_ctxt(cx.backtrace());
         let assert_path = cx.path_all(span, true,
-                                        cx.std_path(&["cmp", helper_name]),
+                                        cx.std_path(&[sym::cmp, Symbol::intern(helper_name)]),
```

Comment:
actually `assert_ty_bounds` looks like it is used multiple times, deduplicate it?

Reply:
Scope creep; let's relegate to a follow-up.